src/Hl7.Fhir.Base/Validation/ValidationContextExtensions.cs

```diff
@@ -85,12 +85,27 @@ public static ValidationContext SetLocation(this ValidationContext ctx, string l
         }
 
         /// <summary>
-        /// Gets the human-readable location for the validation errors from the ValidationContext.
+        /// Alters the ValidationContext to include the human-readable location for the validation errors to use.
         /// </summary>
-        public static string? GetLocation(this ValidationContext ctx) =>
-            ctx.Items.TryGetValue(LOCATION_ITEM_KEY, out var result) && result is string loc ?
-                    loc : null;
+        internal static ValidationContext SetLocation(this ValidationContext ctx, Hl7.Fhir.Serialization.PathStack location)
+        {
+            ctx.Items[LOCATION_ITEM_KEY] = location;
+            return ctx;
+        }
 
+        /// <summary>
+        /// Gets the human-readable location for the validation errors from the ValidationContext.
+        /// </summary>
+        public static string? GetLocation(this ValidationContext ctx)
+        {
+            if (!ctx.Items.TryGetValue(LOCATION_ITEM_KEY, out var result))
+                return null;
+            if (result is Hl7.Fhir.Serialization.PathStack path)
```

Comment:
Should we not have different keys for different things?  Does it make sense to have both the string and the stack?